src/game/components/inventory.rs

```diff
@@ -246,10 +246,16 @@ impl Inventory {
             self.money = self.money - money;
             Ok(money)
         } else {
-            Err(())
```

Comment:
I think what you're looking for here is for the if above to be >= rather than >. This needs to fail if there is not enough money because its used for other purposes like buying things.

```
-  if self.money > money {
+  if self.money >= money {
```